community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/SemanticState.scala

```diff
@@ -186,7 +187,7 @@ import org.neo4j.cypher.internal.compiler.v2_3.SemanticState.ScopeLocation
 case class SemanticState(currentScope: ScopeLocation,
                          typeTable: ASTAnnotationMap[ast.Expression, ExpressionTypeInfo],
                          recordedScopes: ASTAnnotationMap[ast.ASTNode, Scope],
-                         notificationLogger: InternalNotificationLogger = devNullLogger) {
+                         notifications: Set[InternalNotification] = Set.empty) {
```

Comment:
This changes so that we always record notifications even in the case when we use `devNullLogger`. I am not seeing what this change has to do with positioning?